bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java

```diff
@@ -0,0 +1,346 @@
+        T serialized = executeUrl(anUrl, method, aPayload, classOfT, retryAuth, 3);
+        return serialized.getResult();
+    }
+
+    private String sendRequest(Request request) throws FreeboxException {
+        try {
+            ContentResponse response = request.timeout(DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS).send();
+            return new String(response.getContent(), StandardCharsets.UTF_8);
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new FreeboxException("Exception while calling " + request.getURI(), e);
+        }
+    }
+
+    private synchronized <T extends BaseResponse> T executeUrl(URI url, HttpMethod method, @Nullable Object aPayload,
+            @Nullable Class<T> classOfT, boolean retryAuth, int retryCount) throws FreeboxException {
+        logger.debug("executeUrl {} - {} ", method, url);
+        try {
+            Request request = httpClient.newRequest(url).method(method);
+            httpHeaders.entrySet().forEach(entry -> request.header(entry.getKey(), entry.getValue()));
+
+            if (aPayload != null) {
+                String payload = gson.toJson(aPayload);
+                InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+                try (final InputStreamContentProvider contentProvider = new InputStreamContentProvider(stream)) {
+                    request.content(contentProvider, null);
+                }
+            }
+            T serialized = deserialize(classOfT, sendRequest(request));
+            serialized.evaluate();
+            return serialized;
+        } catch (FreeboxException e) {
+            BaseResponse response = e.getResponse();
+            if (response != null) {
+                if (response.getErrorCode() == ErrorCode.INTERNAL_ERROR && retryCount > 0) {
+                    return executeUrl(url, method, aPayload, classOfT, retryAuth, retryCount - 1);
+                } else if (retryAuth && response.getErrorCode() == ErrorCode.AUTHORIZATION_REQUIRED) {
+                    httpHeaders.put(AUTH_HEADER, getLoginManager().openSession(configuration.appToken));
+                    return executeUrl(url, method, aPayload, classOfT, false, retryCount);
+                }
+            }
+            throw e;
+        }
+    }
+
+    private <T extends BaseResponse> T deserialize(@Nullable Class<T> classOfT, String serviceAnswer)
+            throws FreeboxException {
+        try {
+            @Nullable
+            T deserialized = gson.fromJson(serviceAnswer, classOfT != null ? classOfT : BaseResponse.class);
+            if (deserialized != null) {
+                return deserialized;
+            }
+            throw new FreeboxException("Deserialization lead to null object");
+        } catch (JsonSyntaxException e) {
+            if (classOfT != null) {
+                BaseResponse serialized = gson.fromJson(serviceAnswer, BaseResponse.class);
```

Comment:
Take care to catch exceptions.
When running GET - https://mafreebox.freebox.fr:443/api/v8/player/1/api/v8/system an exception occured due to this line because it returned an HTML response rather than a JSON response:
```
13:29:45.498 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:226) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[?:?]
	at com.google.gson.Gson.fromJson(Gson.java:817) ~[?:?]
	at org.openhab.binding.freeboxos.internal.api.ApiHandler.deserialize(ApiHandler.java:331) ~[?:?]
	at org.openhab.binding.freeboxos.internal.api.ApiHandler.executeUrl(ApiHandler.java:303) ~[?:?]
	at org.openhab.binding.freeboxos.internal.api.ApiHandler.execute(ApiHandler.java:268) ~[?:?]
	at org.openhab.binding.freeboxos.internal.api.RestManager.get(RestManager.java:63) ~[?:?]
	at org.openhab.binding.freeboxos.internal.api.player.PlayerManager.getConfig(PlayerManager.java:61) ~[?:?]
	at org.openhab.binding.freeboxos.internal.discovery.FreeboxOsDiscoveryService.discoverPlayer(FreeboxOsDiscoveryService.java:197) ~[?:?]
	at org.openhab.binding.freeboxos.internal.discovery.FreeboxOsDiscoveryService.startScan(FreeboxOsDiscoveryService.java:119) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:386) ~[?:?]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:215) ~[?:?]
	... 17 more
```
Here is the string returned by the API:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>Freebox :: Erreur interne</title>
<link href="/err/err.css" rel="stylesheet" type="text/css" />
</head>

<body>
<div id="info">
    <div id="errorMsg">
      <h3>Erreur interne</h3>
      <p class="desc">Une erreur interne est survenue, réessayez plus tard</p>
    </div>
</div>

</body>
</html>
```
Will see later if these API are really working when the player is ON.